Fast and Flexible GPU Quantization for Transformers.md

```diff
@@ -1,82 +1,193 @@
+A final distinction to be made is how we quantization parameters are shared between elements of our parameters and activations. Throughout this blog, we'll use the following diagram to illustrate a matmul:
 
 ![](_attachments/Pasted%20image%2020230313162138.png)
 
-The simplest approach is to use the same scale factor for all elements of W (and likewise for X). This is **per-tensor** quantization.
+The simplest approach is to use the same scale factor for all elements of $W$ (and likewise for $X$). This is known as **per-tensor** quantization.
 
-It’s also feasible to share quantization parameters between some subgroups of each input matrix. A common choice is to assign a specific scale factor to each column of W. This is **per-channel (aka per-column) quantization**.
+It’s also feasible to share quantization parameters between some subgroups of each input matrix. A popular options is to assign a specific scale factor to each column of $W$, referred to as **per-channel (or per-column) quantization**.
 
-## Theory
-Having covered the basics of quantization, we’ll now look at the important concepts in their implementation.
+## Important Concepts
+With the fundamentals of quantization covered, let's explore the important concepts in its implementation.
 
 ### Calibration
 
-As explained above, calibration is the process of obtaining activation quantization parameters. We pass a few batches of data through the model to measure the distribution over activations.
+Calibration, as mentioned above, involves obtaining activation quantization parameters by passing several batches of data through the model to measure activation distribution.
 
-There are multiple ways of obtaining a clipping range from these activations. These include:
+There are multiple methods to derive a clipping range from these activations, such as:
 
-* Taking a simple min/max
+* Using a simple min/max
 * Minimising KL Divergence between the input and quantized distributions
 * Minimising the Mean-Squared Error between input and quantized distributions
 
-We found that the final approach was most performant (although this may be different for different models).
+We found that the final approach was most performant (although this may vary for different models).
+
+To perform the calibration process, we recommend using TensorRT’s PyTorch Quantization Toolkit [5]. 
+
+### Specifics of  INT8 GEMMs
+The core element of a quantized neural network is INT8 matrix multiplication. Focusing on its details is crucial for an efficient implementation. This section describes these details, and serves as context for the later section describing [Implementation](#Implementation).
+
+We identify two types of INT8 matmul, differentiated by their return type. We'll discuss each of these in turn.
+
+#### i8f16
```

Comment:
I would be tempted to talk about i8i32 here instead, as the i8f16 is quite non-standard (and we touch on it later). What do you think?

Reply:
I find the fusion of Q with X a little confusing, perhaps we explicitly introduce some preceding operator in the diagram.

Reply:
I also find the bias addition a little confusing, perhaps we could do without it?

Reply:
Agree with all these statements. Have added a sentence clarifying that we're omitting the bias for simplicity

Reply:
Have pushed these changes